cmake/extensions.cmake

```diff
@@ -2760,6 +2780,17 @@ function(dt_reg_size var)
   set(${var} ${${var}} PARENT_SCOPE)
 endfunction()
 
+# Internal helper for dt_reg_addr/dt_reg_size; not meant to be used directly
+function(_dt_reg_get_index var path name)
```

Comment:
Don't use `_` as first character of a function.
The `_` is used by CMake to prepend existing functions when they are overloaded, as can be seen by this example:
```
# test.cmake
function(A)
  message("I'm in A")
endfunction()

function(A)
  message("I'm in B")
endfunction()

A()
_A()
```
and running this:
```
$ cmake -P test.cmake
I'm in B
I'm in A
```

Reply:
> `_` is used by CMake to prepend existing functions when they are overloaded

Cool. I didn't know that. Fixed now.